src/UniswapV2Callee.sol

```diff
@@ -110,23 +109,22 @@ contract UniswapV2CalleeDai is UniswapV2Callee {
         uint256 daiToJoin = divup(daiAmt, RAY);
 
         // Do operation and get dai amount bought (checking the profit is achieved)
-        uint256[] memory amounts = uniRouter02.swapExactTokensForTokens(
-                                                  gemAmt,
-                                                  add(daiToJoin, minProfit),
-                                                  path,
-                                                  address(this),
-                                                  block.timestamp
+        uniRouter02.swapExactTokensForTokens(
+            gemAmt,
+            add(daiToJoin, minProfit),
+            path,
+            address(this),
+            block.timestamp
         );
 
-        uint256 daiBought = amounts[1];
-
         // Although Uniswap will accept all gems, this check is a sanity check, just in case
         // Transfer any lingering gem to specified address
         if (gem.balanceOf(address(this)) > 0) {
             gem.transfer(to, gem.balanceOf(address(this)));
         }
 
         // Convert DAI bought to internal vat value of the msg.sender of Clipper.take
+        if (daiJoin.vat().can(address(this), sender) == 0) daiJoin.vat().hope(sender);
```

Comment:
We need to `hope` the clipper so that it has permissions to `move` our dai to the vow.

However, here `sender` isn't the clipper; it's the keeper who called `take`.

So unless I'm missing something, we should be `hope`ing `msg.sender` instead of `sender`.